src/features/author.ts

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+
+type State = {
+  author: User | null
+};
+
+const initialState: State | null = {
+  author: null,
+};
+
+const author = createSlice({
+  name: 'author',
+  initialState,
+  reducers: {
+    setUser: (state, action: PayloadAction<User | null>) => {
+      state.author = action.payload;
+
+      return state;
```

Comment:
There is no need to return state in the setUser reducer. Redux Toolkit's createSlice automatically handles this.